src/coreclr/nativeaot/System.Private.CoreLib/src/System/RuntimeExceptionHelpers.cs

```diff
@@ -192,6 +192,37 @@ public struct ExceptionInformationArray
         }
 
         private static ulong s_crashingThreadId;
+        private static CrashInfo? s_crashInfo;
+        private static object s_crashLockObject = new object();
+
+        internal static CrashInfo SerializeCrashInfo(RhFailFastReason reason, string? message, Exception? exception)
+        {
+            if (s_crashInfo != null)
+            {
+                // If we have already reported a crash, return the existing crash info.
+                return s_crashInfo.Value;
+            }
+
+            lock (s_crashLockObject)
```

Comment:
`FailFast` uses low-tech spin lock currently (`Interlocked.CompareExchange(ref s_crashingThreadId`...). I assume that it is to minimize dependencies. 

Regular `lock` is introducing more dependencies on the crash reporting path, it can execute a lot of code with more opportunities to crash or deadlock if the system state is corrupted.

Would it be better to keep using the low-tech lock here?